readthedocs/rtd_tests/tests/test_project_forms.py

```diff
@@ -424,6 +431,124 @@ def test_external_version_not_in_default_branch_choices(self):
         )
 
 
+class TestProjectPrevalidationForms(TestCase):
+    def setUp(self):
+        # User with connection
+        # User without connection
+        self.user_github = get(User)
+        self.social_github = get(SocialAccount, user=self.user_github)
+        self.user_email = get(User)
+
+    def test_form_prevalidation_email_user(self):
+        form_auto = ProjectAutomaticForm(user=self.user_email)
+        form_manual = ProjectManualForm(user=self.user_email)
+
+        # Test validation errors directly
+        with pytest.raises(ValidationError) as validation_error:
+            form_auto.clean_prevalidation()
```

Comment:
Hrm, I'm not sure you can mix `TestCase` from Django and `pytest` utils. I think we were avoiding this pattern. Since you are using regular `assert` in your tests, I'd recommend you to remove the Django `TestCase` and use only pytest for these tests. cc @stsewd

Reply:
The only case where mixing Django's TestCase and pytest doesn't work is when using `@parametrize`. But, yeah, probably best to stick to one style, and use `self.assert*`.

Reply:
Ah yeah, good call. I'm too used to just relying on pytest :smile: 

I'll update these calls, it should be minor.